### ISSUE TITLE
Feature request: Could you add a typescript type definition to the module?

### ISSUE THREAD

**Issue by agirorn** [brigand/invert-promise#3]
Could you pretty please add a typescript type definition to the module?

It would be very convenient for typescript project if this module provided types. Currently, typescript will complain like this when the module is used.

```
Could not find a declaration file for module 'invert-promise'. 'path-to-project/node_modules/invert-promise/index.js' implicitly has an 'any' type.
      Try `npm install @types/invert-promise` if it exists or add a new declaration (.d.ts) file containing `declare module 'invert-promise';`
```

And the developer has to explicitly set the type to silence it.

**Comment by brigand** [comment]
Yep, should have one, but I'm not sure if there's anything that can be done beyond this.

```ts
// index.d.ts
declare function invertPromise<T>(promise: Promise<T>): Promise<any>;
export default invertPromise;
```

Have any suggestions, or is this what you had in mind? I don't like `any`, but that seems to be how TS types promise rejection values.

**Comment by agirorn** [comment]
I believe this is all it would take to please the typescript gods.

**Comment by brigand** [comment]
Published as 2.0.0. Went with the slightly different `export = invertPromise` which worked with `require()` better in a test project.

**Comment by agirorn** [comment]
Works like a charm, thanks.
